src/Select.tsx

```diff
@@ -16,15 +17,41 @@ export * from './generated/Select.js';
 
 export type SelectReactRendererProps = ReactSimpleRendererProps<SelectElement>;
 
+type SelectRenderer = ComponentType<SelectReactRendererProps>;
+
 export type SelectProps = Partial<Omit<_SelectProps, 'children' | 'renderer'>> &
   Readonly<{
-    children?: ReactNode | ComponentType<SelectReactRendererProps>;
-    renderer?: ComponentType<SelectReactRendererProps> | null;
+    children?: ReactNode | SelectRenderer | Array<ReactNode | SelectRenderer>;
+    renderer?: SelectRenderer | null;
   }>;
 
 function Select(props: SelectProps, ref: ForwardedRef<SelectElement>): ReactElement | null {
+  const children = Children.toArray(props.children as ReactNode[]);
+
+  // Components with slot attribute should stay in light DOM.
+  const slotted = children.filter((child: any) => {
+    return 'props' in child && child.props.slot;
```

Comment:
This will throw an error if a child is a string. You can fix this and avoid use of `any` with `return typeof child === 'object' && 'props' in child && child.props.slot;`

Reply:
Thanks, changed.